test/jdk/javax/accessibility/JSlider/AccessibleAction/JSliderAccessibleAction.java

```diff
@@ -0,0 +1,286 @@
+            System.out.println("changed : " + changeEvent);
+        });
+
+        invalidDecrementBtn = new JButton("Invalid Decrement");
+        invalidDecrementBtn.addActionListener((actionEvent) -> {
+            invalidDecrementCountDownLatch.countDown();
+            accessibleAction.doAccessibleAction(INVALID_DECREMENT);
+        });
+
+        invalidIncrementBtn = new JButton("Invalid Increment");
+        invalidIncrementBtn.addActionListener((actionEvent) -> {
+            invalidIncrementCountDownLatch.countDown();
+            accessibleAction.doAccessibleAction(INVALID_INCREMENT);
+        });
+
+        decrementBtn = new JButton("Decrement");
+        decrementBtn.addActionListener((actionEvent) -> {
+            validDecrementCountDownLatch.countDown();
+            accessibleAction.doAccessibleAction(VALID_DECREMENT);
+        });
+
+        incrementBtn = new JButton("Increment");
+        incrementBtn.addActionListener((actionEvent) -> {
+            accessibleAction.doAccessibleAction(VALID_INCREMENT);
+            validIncrementCountDownLatch.countDown();
+        });
+
+        JPanel buttonPanel = new JPanel(new GridLayout(4, 1));
+        buttonPanel.add(invalidDecrementBtn);
+        buttonPanel.add(invalidIncrementBtn);
+        buttonPanel.add(decrementBtn);
+        buttonPanel.add(incrementBtn);
+        container.add(buttonPanel, BorderLayout.SOUTH);
+
+        jFrame.pack();
+        jFrame.setLocationRelativeTo(null);
+        jFrame.setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);
+        jFrame.setVisible(true);
+    }
+
+    private static boolean setLookAndFeel(String lafName) {
+        try {
+            UIManager.setLookAndFeel(lafName);
+        } catch (UnsupportedLookAndFeelException unsupportedLookAndFeelException) {
+            System.out.println("Ignoring Unsupported laf : " + lafName);
+            return false;
+        } catch (ClassNotFoundException | InstantiationException
+                | IllegalAccessException e) {
+            throw new RuntimeException(e);
+        }
+        return true;
+    }
+
+    public void test() throws AWTException,
+            InterruptedException, InvocationTargetException {
+        Robot robot = new Robot();
+        robot.setAutoDelay(300);
+        robot.waitForIdle();
+
+        List<String> installedLookAndFeels =
+                Arrays.stream(UIManager.getInstalledLookAndFeels())
+                        .map(UIManager.LookAndFeelInfo::getClassName).collect(toList());
+
+        for (String lookAndFeel : installedLookAndFeels) {
+            try {
+                invalidDecrementCountDownLatch = new CountDownLatch(1);
+                invalidIncrementCountDownLatch = new CountDownLatch(1);
+                validDecrementCountDownLatch = new CountDownLatch(1);
+                validIncrementCountDownLatch = new CountDownLatch(1);
+                currentJSliderValue = new AtomicInteger();
```

Comment:
if this is a volatile, than perhaps you do not need an atomic?

Reply:
Fixed